src/ShotTrackerSecondary.java

```diff
@@ -0,0 +1,147 @@
+/**
+ *  * Layered implementations of secondary methods for {@code ShotTracker}.  
+ */
+public abstract class ShotTrackerSecondary implements ShotTracker {
+
+    /**
+     * * Layered implementations of secondary methods for {@code Shot}.
+     */
+    public abstract class Shot implements ShotTracker.Shot {
```

Comment:
This is great. I don't know exactly where this should be implemented, to be honest (according to our discipline). I might recommend creating a separate file called `Shot` and defining it as follows:

```java
public record Shot(int distance, int clubType, int shotType) {}
```

A shot would be immutable, but it makes for a much more pleasurable experience in your code.